thorlcr/graph/thgraphmaster.cpp

```diff
@@ -2760,7 +2762,13 @@ void CMasterGraph::getFinalProgress()
                 if (s>=n)
                     break;
             }
-            throw MakeGraphException(this, 0, "Timeout receiving final progress from slaves - these slaves failed to respond: %s", slaveList.str());
+            if (aborting)
+            {
+                WARNLOG("Timeout receiving final progress from slaves - these slaves failed to respond: %s", slaveList.str());
+                return;
+            }
+            else
+                throw MakeGraphException(this, 0, "Timeout receiving final progress from slaves - these slaves failed to respond: %s", slaveList.str());
```

Comment:
Not new, but given Jake's new code for catching exceptions future code should use some unique exception numbers.